src/entities/utils/calculateProportionalAmounts.ts

```diff
@@ -15,6 +16,7 @@ export function calculateProportionalAmounts(
         tokens: { address: Address; balance: HumanAmount; decimals: number }[];
     },
     referenceAmount: InputAmount,
+    roundDown = true,
```

Comment:
I don't think we should make this optional - if we think this is the best way forward we should just use it and not confuse things with the option.

Reply:
I agree the option may be confusing. I added it because, when rounding down, the output may return even your referenceAmount 1 wei less than the input provided. Anyway, the helper is for us to use and it should return what we need. I'll update over here ✅